lib/sidecar.js

```diff
@@ -1390,4 +1412,96 @@ class Engine extends ReadyResource {
   }
 }
 
+class Collector {
+  async sidecar ({ pid = null, name = 'pear-runtime', flag = '--sidecar' } = {}) {
+    const current = Bare.pid
+    const ids = await this.#lookupProcess({ name, flag })
+    const filtered = ids.filter(id => id !== pid && id !== current)
+    const killed = await this.#kill(filtered)
+    return { killed }
+  }
+
+  async #lookupProcess ({ name = null, flag = null } = {}) {
+    if (!name || !flag) throw new Error('Process name or flag are missing')
+
+    const [sh, args] = isWindows
+      ? ['cmd.exe', ['/c', `wmic process where "executablepath like '%${name}%'" get processid, commandline`]]
+      : ['/bin/sh', ['-c', `ps ax | grep -i -- '${name}' | grep -i -- '${flag}'`]]
+
+    return new Promise((resolve, reject) => {
+      const sp = spawn(sh, args)
+      const output = []
+
+      sp.stdout.on('data', data => {
+        output.push(data.toString().trim())
+      })
+      sp.stderr.on('data', data => { reject(new Error(data.toString())) })
+
+      sp.on('error', err => { reject(err) })
+      sp.on('exit', code => {
+        if (code === 0 || code == null) {
+          const full = output.join('')
+          const lines = full.split('\n')
+          const pids = lines.reduce((acc, line) => {
+            if (line.toLowerCase().includes(flag.toLowerCase())) {
+              const match = line.match(/\d+/)
+              if (match) {
+                const pid = Number(match[0])
+                if (pid !== sp.pid) acc.push(pid)
+              }
+            }
+            return acc
+          }, [])
+          resolve(pids)
+        } else {
+          reject(new Error(`Process exited with code ${code}`))
+        }
+      })
+    })
+  }
+
+  async #kill (ids) {
+    if (ids.length === 0) return []
+
+    const command = isWindows ? 'taskkill' : 'kill'
+    const args = isWindows ? ['/F', '/PID'].concat(ids) : ids
+
+    return new Promise((resolve, reject) => {
+      const sp = spawn(command, args)
+
+      sp.stderr.on('data', data => { console.log(data.toString()) })
```

Comment:
errors a logged to stderr (console.error) not stdout (console.log) 

also, if you want to output the error output in sidecar process from a spawn just inherit STDERR - but only if verbose